$NetBSD$
Overwrite paths and defaults, patch inspired by OpenBSD port version 3.5.2
--- gtk/makefile.orig	2016-10-16 03:10:33.000000000 +0000
+++ gtk/makefile	2016-10-16 20:44:27.385048978 +0000
@@ -33,13 +33,14 @@
 ifdef gnomeprefix
   prefix=$(gnomeprefix)
 else
-  prefix=/usr
+  prefix=$(PREFIX)
 endif
 endif
 datadir=$(prefix)/share
 pixmapdir=$(datadir)/pixmaps
 bindir=$(prefix)/bin
 SYSCONF_PATH=$(prefix)/share/scite
+DOCPATH=$(PREFIX)/share/doc/scite
 
 INSTALL=install
 
@@ -47,10 +48,10 @@
 
 all: $(PROG)
 
-vpath %.h $(srcdir) $(srcdir)/../src $(srcdir)/../../scintilla/include
+vpath %.h $(srcdir) $(srcdir)/../src $(srcdir)/../scintilla
 vpath %.cxx $(srcdir) $(srcdir)/../src
 
-INCLUDEDIRS=-I $(srcdir)/../../scintilla/include -I $(srcdir)/../src
+INCLUDEDIRS=-I $(PREFIX)/include/scintilla -I $(srcdir)/../src -I $(prefix)/include
 ifdef CHECK_DEPRECATED
 DEPRECATED=-DGDK_PIXBUF_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED
 endif
@@ -62,7 +63,9 @@
 CXXTFLAGS=-DNDEBUG -Os $(CXXBASEFLAGS)
 endif
 
-ifndef NO_LUA
+LIBS= -L $(PREFIX)/lib -lscintilla -lscintilla_lexers
+
+ifdef LUA
 LUA_CORE_OBJS = lapi.o lcode.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o \
 		lmem.o lobject.o lopcodes.o lparser.o lstate.o lstring.o \
 		ltable.o ltm.o lundump.o lvm.o lzio.o
@@ -74,7 +77,7 @@
 
 vpath %.c $(srcdir)/../lua/src $(srcdir)/../lua/src/lib
 
-INCLUDEDIRS=-I $(srcdir)/../../scintilla/include -I $(srcdir)/../src -I$(srcdir)/../lua/include
+INCLUDEDIRS+=-I$(srcdir)/../lua/include
 
 .c.o:
 	$(CC) $(INCLUDEDIRS) $(MISLEADING) -DLUA_USE_POSIX $(CXXTFLAGS) $(CFLAGS) -c $< -o $@
@@ -111,16 +114,13 @@
 deps.mak:
 	$(CXX) -MM $(CONFIGFLAGS) $(CXXTFLAGS) $(CXXFLAGS) $(srcdir)/*.cxx $(srcdir)/../src/*.cxx | sed -e 's/\/usr.* //' | grep [a-zA-Z] >$@
 
-# make should be run in ../../scintilla/gtk to compile all the lexers.
-COMPLIB=$(srcdir)/../../scintilla/bin/scintilla.a
-
 $(PROG): SciTEGTK.o GUIGTK.o Widget.o \
 FilePath.o SciTEBase.o FileWorker.o Cookie.o Credits.o SciTEBuffers.o SciTEIO.o StringList.o \
 ExportHTML.o ExportPDF.o ExportRTF.o ExportTEX.o ExportXML.o \
 MatchMarker.o StringHelpers.o \
 PropSetFile.o MultiplexExtension.o DirectorExtension.o SciTEProps.o StyleDefinition.o StyleWriter.o Utf8_16.o \
 	JobQueue.o GTKMutex.o IFaceTable.o $(COMPLIB) $(LUA_OBJS)
-	$(CXX) `$(CONFIGTHREADS)` -rdynamic -Wl,--as-needed -Wl,--version-script $(srcdir)/lua.vers -DGTK $^ -o $@ $(CONFIGLIB) $(LIBDL) -lm -lstdc++
+	$(CXX) `$(CONFIGTHREADS)` $(LIBS) -rdynamic -Wl,--as-needed -Wl,--version-script $(srcdir)/lua.vers -DGTK $^ -o $@ $(CONFIGLIB) $(LIBDL) -lm -lstdc++
 
 # Automatically generate header dependencies with "make deps"
 include deps.mak
@@ -141,7 +141,7 @@
 ifdef gnomeprefix
 	$(INSTALL) -m 755 -d $(DESTDIR)$(datadir)/applications $(DESTDIR)$(pixmapdir)
 	$(INSTALL) -m 755 $(srcdir)/SciTE.desktop $(DESTDIR)$(datadir)/applications/SciTE.desktop
-	$(INSTALL) -m 644 $(srcdir)/Sci48M.png $(DESTDIR)$(pixmapdir)/Sci48M.png
+	$(INSTALL) -m 444 $(srcdir)/Sci48M.png $(DESTDIR)$(pixmapdir)/Sci48M.png
 endif
 
 uninstall:
